riscv/processor.h

```diff
@@ -358,6 +359,8 @@ class processor_t : public abstract_device_t
   bool icache_en;
   bool ic_scr_key_valid;
   std::vector<bool> impl_table;
+  reg_t start_debug = DEBUG_START;
```

Comment:
Don't we need to teach `debug_module_t` and `mmu_t` about the value?

Reply:
No, see the PR comment about debug_module_t, and mmu_t already has a handle to proc. I think having them live in processor_t is just fine.

Reply:
Ahh. Sorry: I hadn't properly read and understood that comment. It's a pity that this leaves things subtly broken if someone tries to use the feature with the built-in debug module. Would it be easy to die with an understandable error if that happens?